src/prefect/agent/kubernetes/agent.py

```diff
@@ -516,7 +516,9 @@ def generate_job_spec_from_environment(
         env[4]["value"] = self.namespace
         env[5]["value"] = str(self.labels)
         env[6]["value"] = str(self.log_to_cloud).lower()
-        env[7]["value"] = config.logging.level
+        env[7]["value"] = self.env_vars.pop(
+            "PREFECT__LOGGING__LEVEL", config.logging.level
+        )
```

Comment:
Probably should not be mutating `self.env_vars`

Reply:
good catch